src/transformStateWithClones.js

```diff
@@ -1,13 +1,43 @@
 'use strict';
 
+// const { meta: action } = require('eslint-plugin-node/lib/rules/no-sync');
+
+//
 /**
  * @param {Object} state
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newArray = [];
+  let newState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        newState = { ...newState, ...action.extraData };
+        newArray.push({ ...newState });
```

Comment:
Consider moving `newArray.push({ ...newState });` outside of the `switch` block, after processing the current action. This will reduce code repetition and improve efficiency.